internal/kafka/internal/migrations/20220511164100_add_expires_at_column.go

```diff
@@ -0,0 +1,71 @@
+package migrations
+
+import (
+	"time"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/environments"
+	"github.com/go-gormigrate/gormigrate/v2"
+	"gorm.io/gorm"
+)
+
+// addExpiresAtColumn adds the expires_at column in the kafka_requests table
+// and sets its value for potentially already existing kafka instances of
+// 'developer' instance_type. The lifespan set for the kafka instance depends
+// on the OCM_ENV value that the migration is run with.
+func addExpiresAtColumn() *gormigrate.Migration {
```

Comment:
Assuming that we still need to persist the expired_at time (I've asked a question in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1027/files#r873909065) 

The way we update existing values does not take into account the fact that a new developer instance might be created during a rollout of this change. Specifically if a developer kafka is created on a existing old pod during a time when the execution of this migration is between line 44-48 with.

We've two options:
- do a phased rollout
- or a use a reconciler to reconciler expires_at time for developer instance.

I think the second option might be better since it also reads the correct values from the configuration and not assumption taken in line 34, 36 and 38 (hardcoding the values). Chances of them changing a low but they could still change.

Reply:
> The way we update existing values does not take into account the fact that a new developer instance might be created during a rollout of this change. Specifically if a developer kafka is created on a existing old pod during a time when the execution of this migration is between line 44-48 with.

That's not accurate at the moment. The calculation for kafka instances expiration is not done from the value of the config file but from the KAFKA_LIFESPAN environment variable (see https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1027#issuecomment-1127830297).
This means we could:
* merge this functionality
* wait until changes are deployed for 2 hours in stage and for 48 hours in production
* merge the changes that update the internal logic to use expire_at to expire the kafka instances

Reply:
> > The way we update existing values does not take into account the fact that a new developer instance might be created during a rollout of this change. Specifically if a developer kafka is created on a existing old pod during a time when the execution of this migration is between line 44-48 with.
> 
> That's not accurate at the moment. The calculation for kafka instances expiration is not done from the value of the config file but from the KAFKA_LIFE_SPAN environment variable at the moment (see [#1027 (comment)](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1027#issuecomment-1127830297)). 

The reconcilers have access to the CLI configurations you mentioned and all the other configurations coming from files (configmap). Generally, if need to access values dynamically that are not reliable accessible at a migration time (e.g data from external services, data that may be modified by a running old pod etc), then a reconciler (temporarily one) is a a fallback option. 

Another option is of course doing a phased rollout which I think is what you’ll go with from the last comment

Reply:
> The reconcilers have access to the CLI configurations you mentioned and all the other configurations coming from files (configmap). Generally, if need to access values dynamically that are not reliable accessible at a migration time (e.g data from external services, data that may be modified by a running old pod etc), then a reconciler (temporarily one) is a a fallback option.

I understand this but I am not sure I am getting what's the point you are trying to explain. The risk of that happening is non-existent due to the logic of kafka expiration calculation does not rely on the new changes that would be applied at the moment.

Reply:
Since the `expires_at` column is used in the presenter, any new kafka created on an old Pod in between the lines I mentioned above won't have the `expires_at`, the information returned for these developers kafkas will be empty.